packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -427,6 +432,70 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate plan metadata on update
+   */
+  async handlePlanUpdate(request: AuthRequest, event: Stripe.Event) {
+    const plan = event.data.object as Stripe.Plan;
+    const products = await this.stripeHelper.allProducts();
+    let product = products.find((p) => p.product_id === plan.product);
+    if (!product) {
+      this.log.error(
+        `handlePlanUpdate - Could not locate Product ${plan.product}`,
+        { item: plan }
+      );
+    } else {
+      const result = subscriptionProductMetadataValidator.validate({
+        ...product.product_metadata,
+        ...plan.metadata,
+      });
+
+      if (result?.error) {
+        const msg = `handlePlanUpdate - Plan "${plan.id}"'s metadata failed validation; After Plan update.`;
+        this.log.error(msg, { item: plan, error: result.error });
+
+        Sentry.withScope((scope) => {
+          scope.setContext('validationError', {
+            error: result.error,
+          });
+          Sentry.captureMessage(
+            `Error validating metadata for plan ${plan.id} and product ${plan.product}`,
+            Sentry.Severity.Error
+          );
+        });
+      }
+    }
+  }
+
+  /**
+   * Validate product metadata on update
+   */
+  async handleProductUpdate(request: AuthRequest, event: Stripe.Event) {
```

Comment:
I noticed that the plan update handler is relying on `allProducts`, which is cached.  We should invalidate that cache here.  Otherwise, the plan update handler could end up using stale products.

Reply:
updated to use `fetchAllProducts`